### PR TITLE
GCW-3462 submit phone number to send pin endpoint

### PR DESCRIPTION
**Pull request by BboyStatix**
## Ticket Link
https://jira.crossroads.org.hk/browse/GCW-3462
## What does this PR do?
Creates the UI that allows user to input their phone number and submit the information to `auth/send_pin`.

### CODE REVIEW

**Review by patrixr — CHANGES_REQUESTED** [review]

File: src/pages/Login/Login.tsx [hunk 1]
```diff
@@ -16,7 +20,11 @@ interface LocationState {
 const Login: React.FC = () => {
   const history = useHistory();
   const location = useLocation<LocationState | undefined>();
+  const [phoneInput, setPhoneInput] = useState("");
+
   const handleClick = () => {
+    const mobile = `+852${phoneInput}`;
+    AuthenticationService.sendPin({ mobile });
```
[the remark this answers is not in the document]
Comment (BboyStatix) [reply src/pages/Login/Login.tsx:27]: Will do so. In another PR

File: src/pages/Login/Login.test.tsx [hunk 2]
```diff
@@ -1,69 +1,206 @@
...
+import { IonButton, IonInput } from "@ionic/react";
+import client from "lib/client/client";
 
 test("renders a login title", () => {
   const { container } = render(<Login />, { wrapper: MemoryRouter });
   expect(container.querySelector("ion-title")).toHaveTextContent(/login/i);
 });
 
-test("renders a go to authenticate button", () => {
+test("renders an ion-item", () => {
```
[the remark this answers is not in the document]
Comment (BboyStatix) [reply src/pages/Login/Login.test.tsx:15]: ### Without IonItem
without
### With IonItem
Screenshot 2021-01-20 at 11 16 07 AM

Therefore, it seems to me that `IonItem` has importance in the sense that it is what results in the UI that we want.

File: src/pages/Login/Login.tsx [hunk 3]
```diff
@@ -32,7 +40,17 @@ const Login: React.FC = () => {
...
+        <IonItem>
+          <IonLabel>+852</IonLabel>
+          <IonInput
+            value={phoneInput}
+            onIonChange={(e) => setPhoneInput(e.detail.value ?? "")}
+            maxlength={8}
+          />
+        </IonItem>
+        <IonButton disabled={phoneInput.length < 8} onClick={handleClick}>
+          Get 4-digit SMS code
```
[the remark this answers is not in the document]
Comment (BboyStatix) [reply src/pages/Login/Login.tsx:52]: Thanks for the heads up. I'll create a ticket for it.

[the remark this answers is not in the document]
Comment (BboyStatix) [reply src/pages/Login/Login.tsx:52]: Ticket created https://jira.crossroads.org.hk/browse/GCW-3480

File: src/pages/Login/Login.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (patrixr) [reply src/pages/Login/Login.tsx:27]: I would consider this a "critical" issue, so definitely to be fixed before merging.
However you can do a rough try/catch+error now and improve in another PR using hooks later

The general premise when reviewing/merging code, is that we don't want to introduce breaking changes, bugs or broken features into the main branch.

File: src/pages/Login/Login.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (patrixr) [reply src/pages/Login/Login.test.tsx:15]: That's not the point. The point is that the spec doesn't make sense. The fact that you use an `IonItem` is part of the implementation, it's not part of the business requirements. So you shouldn't test that

File: src/pages/Login/Login.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrixr) [reply src/pages/Login/Login.tsx:52]: Cool 👍

File: src/pages/Login/Login.test.tsx [hunk 4]
```diff
@@ -1,47 +1,173 @@
...
+  const { container } = render(<Login />, { wrapper: MemoryRouter });
+  expect(container.querySelector("ion-item")).toBeInTheDocument();
+});
```
Comment (patrixr) [inline src/pages/Login/Login.test.tsx:19]: you can remove this

File: src/pages/Login/Login.test.tsx [hunk 5]
```diff
@@ -57,13 +183,56 @@ describe("Clicking go to authenticate button", () => {
...
     });
-    expect(mockHistoryPush).toHaveBeenCalledTimes(1);
 
+    mockHistoryPush.mockRestore();
     mockUseLocation.mockRestore();
   });
 });
+
+describe("On receiving error response from send_pin", () => {
+  it("should log the error to the console", async () => {
```
Comment (patrixr) [inline src/pages/Login/Login.test.tsx:201]: `console.log` isn't really an error management, we probably want an error shown to the user

File: src/pages/Login/test_utils.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+function fillInput(container: HTMLElement, phoneInput: string) {
+  const input = container.querySelector("ion-item > ion-input");
+  act(() => {
+    input!.dispatchEvent(
+      new CustomEvent("ionChange", {
+        detail: { value: phoneInput },
+      })
+    );
+  });
+}
```
Comment (patrixr) [inline src/pages/Login/test_utils.ts:13]: This makes the assumption that the container has `ion-item > ion-input` inside it, so maybe rename the method `fillIonInput` to be clearer, and you can probably do `querySelector('ion-input')` that way `ion-item` won't be required

**Review by patrixr — CHANGES_REQUESTED** [review]
Looks good overall, couple comments

**Pull request merged**